app/models/user.rb

```diff
@@ -561,6 +560,24 @@ def subscribed_to_email_follower_notifications?
     notification_setting.email_follower_notifications
   end
 
+  def last_verification_date
+    email_authorizations
+      .where.not(verified_at: nil)
+      .order(created_at: :desc)
+      .first
+      &.verified_at
+  end
+
+  def reports
+    reporter_feedback_messages.or(affected_feedback_messages).or(offender_feedback_messages)
+  end
+
+  def related_negative_reactions
+    Reaction.article_vomits.where(reactable_id: article_ids)
+      .or(Reaction.comment_vomits.where(reactable_id: comment_ids))
+      .or(Reaction.user_vomits.where(user_id: id))
+  end
```

Comment:
Since `User` is already very big I'm always a bit hesitant to add more to it. Is there any chance these could be scopes/class methods on the associations (`EmailAuthorization` and `Reaction`) instead where pass in a user?

Reply:
Good idea, will see if they feel "Railsy" :D